docs/content/how-to/upgrades.md

```diff
@@ -23,3 +23,67 @@ The HostedControlPlane orchestrates the rollout of the new version of the Contro
 `.spec.release` dictates the version of any particular NodePool.
 
 A NodePool will perform a Replace/InPlace rolling upgrade according to `.spec.management.upgradeType`.
+
+### Replace Upgrades
+
+Replace upgrades are backed by MachineDeployments. When an update is triggered, an ephemeral Ignition Server instance created via running the Machine-Config-Operator in bootstrap mode generates the new set of configuration, which is then used to boot new RHCOS instances. Old Nodes are deprovisioned as part of this process. The NodePool Controller in the Hypershift Operator reconciles the MachineDeployment and performs a rolling upgrade based on Strategy (RollingUpdate|OnDelete). If RollingUpdate is specified, you can optionally set maxUnavailable and maxSurge to tweak the upgrade.
+
+### InPlace Upgrades
+
+InPlace upgrades are backed by MachineSets and controllers running as part of the Control Plane Operator's Hosted Cluster Config Operator. The configuration is generated in the same fashion as Replace Upgrades, but no new machines are created. The HCCO's Inplaceupgrader, upon seeing the changes, will start a rolling upgrade much like self-driving OpenShift's Machine Config Operator:
+
+1. the HCCO's Inplaceupgrader creates a set of manifests, including a ConfigMap (used to store the updated MachineConfig to be mounted into the MachineConfigDaemon pods), a Namespace/ClusterRole/ClusterRoleBinding (used by the MachineConfigDaemon), and a temporary daemonset running the MachineConfigDaemon in a special Hypershift mode.
```

Comment:
Same as above, let's focus in API input/output and best practices, internals can be elaborated [in the enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/hypershift/node-lifecycle.md).